packages/liquidator/index.js

```diff
@@ -53,24 +61,52 @@ async function run({
     const getTime = () => Math.round(new Date().getTime() / 1000);
 
     // Load unlocked web3 accounts and get the networkId.
-    const [accounts, networkId] = await Promise.all([web3.eth.getAccounts(), web3.eth.net.getId()]);
+    const [detectedContract, accounts, networkId] = await Promise.all([
+      findContractVersion(empAddress, web3),
+      web3.eth.getAccounts(),
+      web3.eth.net.getId()
+    ]);
+
+    // Append the contract version and type to the liquidatorConfig, if the liquidatorConfig does not already contain one.
+    if (!liquidatorConfig) liquidatorConfig = {};
```

Comment:
What if the passed in `liquidatorConfig.contractVersion` and type do not match the `detectedContract`'s version and type?

Reply:
then it defaults to the passed in version. that's the preferred version.